pkg/certwatcher/certwatcher.go

```diff
@@ -93,7 +92,7 @@ func (cw *CertWatcher) Start(ctx context.Context) error {
 				files.Delete(f)
 			}
 			return true, nil
-		}); err != nil {
+		}); errors.Is(err, context.DeadlineExceeded) {
```

Comment:
I think we should always return an err here if we encountered one, not only when its DeadlineExceeded

Reply:
will change to `wait.Interrupted(err)`. the context func does not seem to any return errors.